packages/typespec-client-generator-core/test/types/model-types.test.ts

```diff
@@ -1502,4 +1502,71 @@ describe("typespec-client-generator-core: model types", () => {
     ok(inputModel);
     strictEqual(inputModel.usage, UsageFlags.Input | UsageFlags.Xml);
   });
+
+  it("check bodyParam for @multipartBody", async function () {
+    await runner.compileWithBuiltInService(`
+        model Address {
+          city: string;
+        }
+        model MultiPartRequest{
+          id?: HttpPart<string>;
+          profileImage: HttpPart<bytes>;
+          address: HttpPart<Address>;
+          picture: HttpPart<File>;
+        }
+        @post
+        op upload(@header contentType: "multipart/form-data", @multipartBody body: MultiPartRequest): void;
+        `);
+    const formDataMethod = runner.context.sdkPackage.clients[0].methods[0];
```

Comment:
The test looks good to me. 
Would like to know the behavior of inheritance case, do we want to add a test case to cover case like below?
```ts
model InheritFile extends File {
  contentType?: string;
  filename: string;
}

op uploadHttpPart(
  @header contentType: "multipart/form-data",
  @multipartBody body: {
    fileData1: HttpPart<InheritFile>;
  },
): OkResponse;
```

Reply:
this case is to make sure TCGC parse @multipartBody correctly and inheritance shall not influence the result.